File: week_1/day_1_arrays/kadane_follow_up_questions/return_the_subarray_indices.js
function maxSubArrayWithIndices(nums) {
    let maxSum = nums[0], currSum = nums[0];
    let start = 0, end = 0, tempStart = 0;

    for (let i = 1; i < nums.length; i++) {
        if (nums[i] > currSum + nums[i]) {
            currSum = nums[i];
            tempStart = i;
        } else {
            currSum += nums[i];
        }

        if (currSum > maxSum) {
            maxSum = currSum;
            start = tempStart;
            end = i;
        }
    }

    return { maxSum, start, end };
}

const result = maxSubArrayWithIndices(
  [-2, 1, -3, 4, -1, 2, 1, -5, 4]
);
console.log(result);

File: week_1/day_1_arrays/kadane_follow_up_questions/return_the_subarray_indices.py
def maxSubArrayWithIndices(nums):
    max_sum = curr_sum = nums[0]
    start = end = temp_start = 0

    for i in range(1, len(nums)):
        if nums[i] > curr_sum + nums[i]:
            curr_sum = nums[i]
            temp_start = i
        else:
            curr_sum += nums[i]

        if curr_sum > max_sum:
            max_sum = curr_sum
            start = temp_start
            end = i

    return max_sum, start, end


nums = [-2, 1, -3, 4, -1, 2, 1, -5, 4]
print(maxSubArrayWithIndices(nums))
